app/concepts/screenings/use_cases/find_available_seats.rb

```diff
@@ -0,0 +1,34 @@
+module Screenings
+  module UseCases
+    class FindAvailableSeats 
+
+      def initialize(screening_id)
+        @reservations = Reservations::UseCases::FetchReservationsForScreening.new.call(screening_id)
+        screening = Screenings::Repository.new.find_by(screening_id)
+        cinema_hall = CinemaHalls::Repository.new.find_by(screening.cinema_hall_id)
+        @seats = cinema_hall.seats
+      end
+
+      def call
+        find_reserved_seats
+        find_available_seats
+        @available_seats
+        
+      end
+
+      def find_reserved_seats
+        @taken_seats =Array.new
+        @reservations.map do |reservation|
+          reservation.tickets.map do |ticket|
+            @taken_seats.push(ticket.seat)
+          end
+        end
```

Comment:
I think you have there N+1 query problem, it would be best to preload tickets before executing this query.

Reply:
Could you give me a hint how to preload tickets?  Should I first save them all to the variable?

Reply:
[Here](https://blog.saeloun.com/2020/01/08/activerecord-database-performance-n-1-includes-preload-eager-load-pluck.html) you have a really good article about it. I know that on a daily summary I've described solution a little bit differently, but this is also a really good solution to solve this problem. Consider `includes` method usage `reservation.includes(:tickets).tickets.map do |ticket|`